src/providers/mistral-ai/fimComplete.ts

```diff
@@ -0,0 +1,190 @@
+import { MISTRAL_AI } from '../../globals';
+import {
+    FimCompletionResponse,
+    ErrorResponse,
+    ProviderConfig,
+    MistralFinishReason
+  } from '../types';
+
+  import {
+    generateErrorResponse,
+    generateInvalidProviderResponseError,
+  } from '../utils';
+
+  export const MistralAIFimCompleteConfig: ProviderConfig = {
+    prompt: {
+        param: 'prompt',
+        required: true,
+        default: '',
+    },
+    suffix: {
+        param: 'suffix',
+        required: false,
+        default: '',
+    },
+    model: {
+      param: 'model',
+      required: true,
+      default: 'codestral-latest',
+    },
+    temperature: {
+      param: 'temperature',
+      default: 0.7,
+      min: 0,
+      max: 1,
+    },
+    top_p: {
+      param: 'top_p',
+      default: 1,
+      min: 0,
+      max: 1,
+    },
+    max_tokens: {
+      param: 'max_tokens',
+      default: null,
+      min: 1,
+    },
+    min_tokens: {
+        param: 'max_tokens',
+        default: null,
+        min: 1,
+      },
+    stream: {
+      param: 'stream',
+      default: false,
+    },
+    seed: {
+      param: 'random_seed',
+      default: null,
+    },
+    stop: {
+        param: 'stop',
+        default: []
+    }
+};
+
+interface MistralAIFimCompleteResponse extends FimCompletionResponse {
+    id: string;
+    object: string;
+    created: number;
+    model: string;
+    usage: {
+      prompt_tokens: number;
+      completion_tokens: number;
+      total_tokens: number;
+    };
+}
+
+export interface MistralAIErrorResponse {
+    object: string;
+    message: string;
+    type: string;
+    param: string | null;
+    code: string;
+}
+
+
+interface MistralAIStreamChunk {
+    id: string;
+    object: string;
+    created: number;
+    model: string;
+    choices: {
+      delta: {
+        role?: string | null;
+        content?: string;
+      };
+      index: number;
+      finish_reason: string | null;
+    }[];
+  }
+
+
+  function transformMistralFinishReasonToOpenAIFinishReason(finish_reason: MistralFinishReason | string): string {
+    switch (finish_reason) {
+        case MistralFinishReason.STOP:
+            return 'stop';
+        case MistralFinishReason.LENGTH:
+            return 'length';
+        case MistralFinishReason.MODEL_LENGTH:
+            return 'model_length';
```

Comment:
model_length is not an accepted finish reason from openai schema, you are returning the same enum as you recieve??
Please check the OpenAI documentaition to see what are the finish_reasons
![Uploading image.png…]()